lib/dsc-lib/src/util.rs

```diff
@@ -232,6 +233,29 @@ pub fn resource_id(type_name: &str, name: &str) -> String {
     result
 }
 
+pub fn canonicalize_which(executable: &str, cwd: Option<&str>) -> Result<String, DscError> {
+    // Use PathBuf to handle path separators robustly
+    let mut executable_path = PathBuf::from(executable);
+    if cfg!(target_os = "windows") && executable_path.extension().is_none() {
+        executable_path.set_extension("exe");
+    }
+    let mut executable = executable_path.to_string_lossy().to_string();
+    if which(&executable).is_err() && !Path::new(&executable).is_absolute() {
```

Comment:
Since `PathBuf` derefs to `Path`, did we need to go through the string conversion? Actually, since it has `is_absolute` did we need to ditech `executable_path` at all? It must have to do with the lossy conversion. I'll go look at `which`.